packages/edit-post/src/components/meta-boxes/index.js

```diff
@@ -1,25 +1,36 @@
+/**
+ * External dependencies
+ */
+import { map } from 'lodash';
+
 /**
  * WordPress dependencies
  */
 import { withSelect } from '@wordpress/data';
+import { Fragment } from '@wordpress/element';
 
 /**
  * Internal dependencies
  */
 import MetaBoxesArea from './meta-boxes-area';
+import MetaBoxVisibility from './meta-box-visibility';
 
-function MetaBoxes( { location, isActive } ) {
-	if ( ! isActive ) {
-		return null;
-	}
-
-	return <MetaBoxesArea location={ location } />;
+function MetaBoxes( { location, isVisible, metaBoxes } ) {
+	return (
+		<Fragment>
+			{ map( metaBoxes, ( { id } ) => (
+				<MetaBoxVisibility key={ id } id={ id } />
+			) ) }
+			{ isVisible && <MetaBoxesArea location={ location } /> }
```

Comment:
Is it necessary to render this component _at all_ if `! isVisible`? i.e. an `ifCondition`.

Reply:
It's necessary to handle the visibility.